subprojects/core-api/src/integTest/groovy/org/gradle/internal/scripts/DefaultScriptFileResolverTest.groovy

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.scripts
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class DefaultScriptFileResolverTest extends AbstractIntegrationSpec {
+
+    def "when multiple build scripts are present, the resolved will report a warning"() {
+        given:
+        enableProblemsApiCheck()
+        file("build.gradle").touch()
```

Comment:
I think this should instead test for all possible pairs of duplicate files (groovy + kts, groovy + dcl, kts + dcl).